### PR TITLE
Only download JS zip when integTest is running

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

*Issue #, if available:*
#333

*Description of changes:*
Only download JS zip when integTest is running

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -268,10 +265,23 @@ testClusters.integTest {
             debugPort += 1
         }
     }
-    plugin(provider({
-        new RegularFile() {
-            @Override
-            File getAsFile() { fileTree(job_scheduler_resource_folder).getSingleFile() }
+
+    plugin(provider(new Callable<RegularFile>(){
```
Comment (thalurur) [inline build.gradle:269]: out of curiosity: why not just call the getPluginResource before trying to load the file?

Comment (peterzhuamazon) [reply build.gradle:269]: Tried to do that initially but the call still happens during assemble.

**Comment by peterzhuamazon** [comment]
Interesting that bwc test success on my local while instantly fail in github actions.

File: build.gradle (hunk 1 above)
Comment (peterzhuamazon) [reply build.gradle:269]: Have some research and figure out the correct location now.
Thanks.

Comment (peterzhuamazon) [reply build.gradle:270]: Some more testing and that method I thought runs didnt.
Revert back.

**Review by dblock — APPROVED** [review]

**Review by downsrob — APPROVED** [review]

**Pull request merged**